integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/BaseFhirR3IntegrationTest.java

```diff
@@ -112,4 +117,98 @@ protected void describeMismatchSafely(Bundle.BundleEntryComponent item, Descript
+			        .filter(it -> it instanceof Observation).map(it -> (Observation) it).collect(Collectors.toList());
+			
+			Set<String> codeList = new HashSet<>();
+			
+			for (int var = 0; var < observations.size(); var++) {
+				String currentConcept = observations.get(var).getCode().getCodingFirstRep().getCode();
+				
+				//check if Bundle is returned grouped code wise
+				// if an observation arrives whose concept wise list is already created, then that means this observation is
+				//out of place
+				//so bundle is not returned grouped wise
+				if (codeList.contains(currentConcept)) {
+					mismatchDescription.appendText("an Observation which was not grouped correctly")
+					        .appendValue(observations.get(var).getId());
+					return false;
+				} else {
+					
+					codeList.add(currentConcept);
+					
+					String currentDateTimeType = observations.get(var).getEffectiveDateTimeType().toString();
+					
+					int distinctObsDateTime = 1;
+					
+					if (var == observations.size() - 1) {
+						return true;
+					}
+					
+					String nextConcept = observations.get(var + 1).getCode().getCodingFirstRep().getCode();
+					String nextDateTimeType = observations.get(var + 1).getEffectiveDateTimeType().toString();
+					
+					while (nextConcept.equals(currentConcept)) {
+						//if nextDatetime is greater than currentDateTime, then the list is not sorted in decreasing order, which was required
+						if (currentDateTimeType.compareTo(nextDateTimeType) < 0) {
+							mismatchDescription.appendText("an Observation which is not placed in sorted order")
+							        .appendValue(observations.get(var + 1).getId());
+							return false;
+						}
+						
+						if (!currentDateTimeType.equals(nextDateTimeType)) {
+							distinctObsDateTime++;
+						}
+						var++;
+						
+						if (var + 1 == observations.size()) {
+							//if count of distinct obsDatetime is within max
+							if (distinctObsDateTime <= max) {
+								return true;
+							}
+							
+							mismatchDescription
+							        .appendText("a code group which contains more observations than specified in max")
+							        .appendValue(currentConcept);
+							return false;
+						}
+						
+						nextConcept = observations.get(var + 1).getCode().getCodingFirstRep().getCode();
+						
+						currentDateTimeType = nextDateTimeType;
+						nextDateTimeType = observations.get(var + 1).getEffectiveDateTimeType().toString();
+					}
+					
+					if (distinctObsDateTime > max) {
+						mismatchDescription.appendText("a code group which contains more observations than specified in max")
+						        .appendValue(currentConcept);
+						return false;
+					}
+				}
+			}
+			
+			return true;
+		}
+		
+		@Override
+		public void describeTo(Description description) {
+			description.appendText(
+			    "Result is grouped code wise, sorted from recent to older and distinct obsDatetime is within value of max");
```

Comment:
"Result is grouped by concept and sorted from most recent to oldest with up to $max distinct observation times" or something